compiler/src/iree/compiler/Codegen/Common/CPU/CPULowerToUKernels.cpp

```diff
@@ -477,10 +477,12 @@ static uint32_t
 getFlagForUserAndOperandTypes(IREE::Encoding::EncodingAttr encoding,
                               ArrayRef<Type> operandTypes) {
   // There are currently no batch_mmt4d ukernels, so check for no batch
-  // dimension.
+  // dimension. We should not reach to the case because only VMVX backend cares
+  // the dynamic inner tile sizes and the batch gemm data-tiling is disabled in
+  // the first place, if ukernel is enabled.
   auto cDims = IREE::Encoding::getEncodingContractionDims(encoding);
-  if (failed(cDims) || !cDims->batch.empty() || operandTypes.size() != 3) {
-    return IREE_UK_FLAG_QUERY_TILE_SIZES_OPERATION_NONE;
+  if (succeeded(cDims)) {
```

Comment:
This condition seems like it is triggerable by IR, even if it's IR that we deem ill-formed. We should not allow even an ill-formed input program to trigger an assertion. That is why we normally reserve assertions for local self-checks of C++ code, enforcing contracts between implementation pieces, not going through user input (IR).

If you need to propagate an error, maybe change the return type of this function to `std::optional<uint32_t>` and propagate that to the CPULowerToUKernels pass where it can become a pas failure.

Reply:
Yes, you're right. It can be triggered by IR. How about we return the code only when the contraction dimensions are inferrable?

Reply:
sg - since it only affects VMVX, whatever passes tests.